flow/layers/clip_rrect_layer.cc

```diff
@@ -37,8 +37,7 @@ void ClipRRectLayer::Paint(PaintContext& context) {
   TRACE_EVENT0("flutter", "ClipRRectLayer::Paint");
   FTL_DCHECK(!needs_system_composite());
 
-  SkAutoCanvasRestore save(&context.canvas, false);
-  context.canvas.saveLayer(&paint_bounds(), nullptr);
+  Layer::AutoSaveLayer save(context, paint_bounds(), nullptr);
```

Comment:
SkAutoCanvasRestore(canvas, false) doesn't do an SkCanvas::save().  The SkCanvas::save() happens within the saveLayer() call